python/rmm/_lib/device_buffer.pxd

```diff
@@ -36,10 +42,10 @@ cdef extern from "rmm/device_buffer.hpp" namespace "rmm" nogil:
 cdef class DeviceBuffer:
     cdef unique_ptr[device_buffer] c_obj
 
-    # Holds a reference to the DeviceMemoryResource used for allocation.
-    # Ensures the MR does not get destroyed before this DeviceBuffer. `mr` is
-    # needed for deallocation
-    cdef DeviceMemoryResource mr
+    # Holds a reference to the device_memory_resource used for allocation.
+    # Ensures the memory resource does not get destroyed before
+    # this DeviceBuffer as it is needed for deallocation.
```

Comment:
In the future,  we may be tempted to replace this again with a reference to the `DeviceMemoryResource` instead of a `shared_ptr` attribute. I think we should extend this comment to include information about  why that would  be a  bad idea.